subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/suite/filter/FilterFactory.kt

```diff
@@ -0,0 +1,167 @@
+package com.avito.instrumentation.suite.filter
+
+import com.avito.instrumentation.configuration.InstrumentationFilter
+import com.avito.instrumentation.configuration.InstrumentationFilter.FromRunHistory
+import com.avito.report.ReportsFetchApi
+import com.avito.report.model.ReportCoordinates
+import com.avito.report.model.SimpleRunTest
+import java.io.File
+
+class FilterFactory(
+    private val filterData: InstrumentationFilter.Data,
+    private val impactAnalysisResult: File?,
+    private val reportCoordinates: ReportCoordinates,
+    private val reportsFetchApi: ReportsFetchApi
+) {
+
+    fun createInitialFilter(): TestsFilter {
+        val filters = mutableListOf<TestsFilter>()
+        filters.add(ExcludeBySdkFilter)
+        filters.addAnnotationFilters()
+        filters.addSourceCodeSignaturesFilters()
+        filters.addSourcePreviousSignatureFilters()
+        filters.addSourceReportSignatureFilters()
+        filters.addImpactAnalysisFilter()
+        return CompositionFilter(
+            filters
+        )
+    }
+
+    fun createRerunFilter(): TestsFilter {
+        val filters = mutableListOf<TestsFilter>()
+        filters.addSourcePreviousSignatureFilters()
+        return CompositionFilter(
+            filters
+        )
+    }
+
+    private fun MutableList<TestsFilter>.addAnnotationFilters() {
+        if (!filterData.fromSource.annotations.included.isNullOrEmpty()) {
```

Comment:
Include mean that test will be executed only if it matches included and doesn't match excluded
If include is empty it means that everything includes by default.
That's why I didn't add include filter when it's empty.